### PR TITLE
only log `done` when the delivery succeeded; log an error upon exception.

### PR DESCRIPTION
**Pull request by zdAlexM**
When Net::SMTP raises an exception, this logger didn't previously have
anything to say. Now, it logs that outcome.

### CODE REVIEW

File: lib/action_mailer/logged_smtp_delivery.rb [hunk 1]
```diff
@@ -13,17 +13,23 @@ def initialize(settings)
...
-    self.response = super
-    log mail, "done #{response.inspect}"
+    begin
+      self.response = super
+    rescue StandardError => e
+      logger.error("#{mail.message_id} exception #{e.inspect}")
+      raise
+    else
+      log mail, "done #{response.inspect}"
+    end
```
Comment (zenspider) [inline lib/action_mailer/logged_smtp_delivery.rb:32]:
```suggestion
    self.response = super
    log mail, "done #{response.inspect}"
  rescue => e
    logger.error("#{mail.message_id} exception #{e.inspect}")
    raise
```

File: test/logged_smtp_delivery_test.rb [hunk 2]
```diff
@@ -128,5 +128,26 @@ def without_file_logger
...
+          end
+        end
+      end
+      it 'logs any StandardError being thrown by the delivery method' do
+        assert_raises(StandardError) { TestMailer.welcome.deliver_now }
+        assert_includes log.string, '12345@example.com exception #<StandardError: kaboom>'
+      end
+      after do
+        class Net::SMTP
+            alias_method :rcptto_list, :orig_rcptto_list
```
Comment (zenspider) [inline test/logged_smtp_delivery_test.rb:148]: indent is wrong

File: test/logged_smtp_delivery_test.rb [hunk 3]
```diff
@@ -128,5 +128,26 @@ def without_file_logger
...
+
+    describe 'delivery failures' do
+      require 'net/smtp'
+      before do
+        class Net::SMTP
+          alias_method :orig_rcptto_list, :rcptto_list
+          def rcptto_list(*)
+            raise StandardError.new("kaboom")
+          end
+        end
```
Comment (zenspider) [inline test/logged_smtp_delivery_test.rb:140]: this honestly feels a little wierd. Why not use the FakeSMTP or stubbing to raise on this method call?

Comment (zdAlexM) [reply test/logged_smtp_delivery_test.rb:140]: It is admittedly a little weird, but I couldn't get the advertised minitest `Net::SMTP.mock :rcptto_list` call to work; it said there was no such method to mock.

Also, `FakeSMTP` is unused as of commit e5ad5c3ec86a141a1f684f883bc48b4fec071186 wherein the `adaptor` mechanism went away, and the fake `mailcrate` smtpd that this test uses isn't configurable enough to allow the error to come "naturally" from the SMTP transaction.

**Review by sircpl — APPROVED** [review]

**Pull request merged**